code/data_structures/src/tree/binary_tree/avl_tree/avl__tree.py

```diff
@@ -0,0 +1,245 @@
+    if outputdebug:
+        print msg
+
+class Node():
+    def __init__(self, key):
+        self.key = key
+        
+class AVLTree():
+    def __init__(self, *args):
+        self.node = None 
+        self.height = -1  
+        self.balance = 0; 
+        
+        if len(args) == 1: 
+            for i in args[0]: 
+                self.insert(i)
+     def height(self):
+        if self.node: 
+            return 0 
+    
+    def is_leaf (self):
+        return self.height == 0
+    
+    def insert(self, key):
+        tree = self.node
+        
+        newnode = Node(key)
+        
+        if tree == None:
+               self.node = newnode 
+            self.node.left = AVLTree() 
+            self.node.right = AVLTree()   
+			
+        elif key < tree.key: 
+            self.node.left.insert(key)
+            
+        elif key > tree.key: 
+            self.node.right.insert(key)
+        
+        else: 
+            
+        self.rebalance() 
+        
+    def rebalance(self):
```

Comment:
Ok. Could you please the clarify the distinction here between a Node and a Tree? Your code seems to overlap the definitions.